### PR TITLE
Add additional combinators for resoning in monoidal categories.

### PR DESCRIPTION
**Pull request by sstucki**
Just some extra reasoning combinators. (Preparation for another upcoming PR.)

### CODE REVIEW

File: src/Categories/Category/Monoidal/Reasoning.agda [hunk 1]
```diff
@@ -137,3 +137,81 @@ split₂ˡ {f = f} {g} {h} = begin
...
+
+module _ {X Y Z} {f : Y ⇒ Z} {g : X ⇒ Y} {h : X ⇒ Z} (fg≈h : f ∘ g ≈ h) where
+
+  infixr 4 merge₁_⟩∘⟨_ merge₂_⟩∘⟨_
+  infixl 5 _⟩∘⟨merge₁_ _⟩∘⟨merge₂_
+
+  merge₁_⟩∘⟨_ : ∀ {V W} {i j : V ⇒ X ⊗₀ W} → i ≈ j →
+                f ⊗₁ C.id ∘ g ⊗₁ C.id ∘ i ≈ h ⊗₁ C.id ∘ j
+  merge₁_⟩∘⟨_ {_} {_} {i} {j} i≈j = begin
+    f ⊗₁ C.id ∘ (g ⊗₁ C.id ∘ i)    ≈⟨ C.sym-assoc ⟩
```
Comment (JacquesCarette) [inline src/Categories/Category/Monoidal/Reasoning.agda:190]: It looks like all these proofs are the formal reverse of the ones for split - perhaps they could be simplified?

**Review by JacquesCarette — CHANGES_REQUESTED** [review]

File: src/Categories/Category/Monoidal/Reasoning.agda (hunk 1 above)
Comment (sstucki) [reply src/Categories/Category/Monoidal/Reasoning.agda:190]: Of course. Silly me. I'll fix these.

**Comment by sstucki** [comment]
I updated the proofs of the `merge` combinators as per @JacquesCarette's request and cleaned up the module a bit while I was at it.

**Review by JacquesCarette — APPROVED** [review]

**Comment by sstucki** [comment]
Should I merge?

**Comment by JacquesCarette** [comment]
If it's been approved and the build is clean, yes, go ahead.  I've done it for this one.

**Pull request merged**